analytical_engine/apps/boundary/edge_boundary.h

```diff
@@ -53,16 +53,18 @@ class EdgeBoundary : public AppBase<FRAG_T, EdgeBoundaryContext<FRAG_T>>,
     vid_t gid;
     vertex_t u;
     bool no_target_set = true;
-    for (const auto& node : source_array.GetArray()) {
-      if (frag.Oid2Gid(dynamic_to_oid<oid_t>(node), gid)) {
+    for (auto& node : source_array) {
+      dynamic::Value oid(node);
+      if (frag.Oid2Gid(dynamic_to_oid<oid_t>(oid), gid)) {
```

Comment:
Looks strange about the `dynamic::Value oid(node)`, why we need such a (implicit) copy as we will obtain a vanilla `oid_t` value and the reference won't be used any more?

Reply:
Because in `for(auto& node: source_array)` , the node is `rapidjson::Value`, not `dynamic::Value`.

Reply:
Could we have a overload for `dynamic_to_oid<oid_t>(rapidjson::Value const &)` as well?

Reply:
Fixed.